components/kubeflow-pipeline/pipeline_components/analytics-server/tests/test_fedhist.py

```diff
@@ -0,0 +1,52 @@
+"""
+
+Federated Histogram Utilities : Functional tests
+
+Katulu GmbH 
+(c) 2022 
+
+"""
+import numpy as np
+
+from flwr_analytics_server.fedhist import HistogramData, aggregate_histograms
+
+
+def run_two_client_example(_rtol=1e-05, _atol=1e-08) -> None:
+    """Run two client histogram construction."""
+    #
+    # Two client : histogram constuction
+    #
+    # Test data
+    #
+    np.random.seed(4242)
+    global_data = np.random.gamma(shape=2.0, scale=2.0, size=1000)
+    local_data_0 = global_data[0:500]
+    local_data_1 = global_data[500:1000]
+    #
+    # Compute local histograms
+    #
+    client_0_histogram = np.histogram(local_data_0, bins=90, range=(0.14, 23.0))
+    client_1_histogram = np.histogram(local_data_1, bins=90, range=(0.14, 23.0))
+    #
+    # Aggregation for global histogram
+    #
+    global_histogram = aggregate_histograms(
+        [
+            HistogramData(client_0_histogram[0], client_0_histogram[1]),
+            HistogramData(client_1_histogram[0], client_1_histogram[1]),
+        ]
+    )
```

Comment:
I think we should be able to test the actual `HistogramProvider.aggregate_histograms` by instantiating it instead of calling the private `aggregate_histograms` function: 

```suggestion
    h1 = HistogramProvider()
    h1.add_client_data({"couts": client_0_histogram[0], "bins": client_0_histogram[1]})
    h1.add_client_data({"couts": client_1_histogram[0], "bins": client_1_histogram[1]})

    h1.aggregate()
```

Reply:
Good point, we should test the providers instead of only testing their algorithms.